src/functions.php

```diff
@@ -285,6 +341,10 @@ function (mixed $throwable) use (&$rejected, &$rejectedThrowable, &$fiber): void
         throw $rejectedThrowable;
     }
 
+    if ($lowLevelFiber !== null) {
+        fiberMap()->attachPromise($lowLevelFiber, $promise);
```

Comment:
Seems reasonable. Makes me wonder why we don't have a `detachPromise()` method though. I suppose the following example would produce a lot of garbage references until the `Fiber` is eventually freed?

```php
async(function () {
    for($i = 0; $i < 1_000_000; ++$i) {
        $promise = new Promise(function ($resolve) {
            Loop::futureTick(fn() => $resolve(null));
        });
        await($promise);
    }
})();